pkgs/applications/misc/cotp/default.nix

```diff
@@ -0,0 +1,33 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+  AppKit,
+  libxcb,
+}:
+rustPlatform.buildRustPackage rec {
+  name = "cotp";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "replydev";
+    repo = "cotp";
+    rev = "v${version}";
+    sha256 = "sha256-SbS+ch7/45kZb49jW2mnRWQruLrfrNfeZFqEPHQKGUg=";
+  };
+
+  cargoSha256 = "sha256-2lIR3K4/hr4XSmNGFd/dhwoFOtEB9KSnUrZkcaCyc9k=";
+
+  buildInputs =
+    lib.optionals stdenv.isLinux [libxcb]
+    ++ lib.optionals stdenv.isDarwin [AppKit];
+
+  meta = with lib; {
+    homepage = "https://github.com/replydev/cotp";
+    description = "Trustworthy, encrypted, command-line TOTP/HOTP authenticator app with import functionality";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [davsanchez];
+    platforms = ["x86_64-linux" "aarch64-darwin"];
```

Comment:
Why specifically these platforms?

Reply:
These are the ones where I have built and tested. Not sure if it would be good practice to include all without having checked on my own, but I do assume it would work for both `aarch64-linux` and `x86_64-darwin` as well.

Reply:
In that case you should remove that line, this will default to `platforms.unix`. 

There's a CI bot called @ofborg which will build for specified platforms in the scope of your PR. You can check the build status in the "Checks" section to confirm that all builds succeeded, that way you don't have to necessarily have a direct access to all platforms. If something failed and you don't know how to fix that, you can just set `meta.broken` and specify the failing platform or ask around for help.